Modules/posixmodule.c

```diff
@@ -14962,6 +14962,326 @@ os_waitstatus_to_exitcode_impl(PyObject *module, PyObject *status_obj)
+        {
+            result = !(info.FileAttributes & FILE_ATTRIBUTE_DIRECTORY);
+        }
+        else {
+            result = 0;
+        }
+        if (close_file) {
+            CloseHandle(hfile);
+        }
+    }
+    else {
+        STRUCT_STAT st;
+        switch (GetLastError()) {
+        case ERROR_ACCESS_DENIED:
+        case ERROR_SHARING_VIOLATION:
+        case ERROR_CANT_ACCESS_FILE:
+        case ERROR_INVALID_PARAMETER:
+            if (STAT(_path.wide, &st)) {
+                result = 0;
+            }
+            else {
+                result = S_ISREG(st.st_mode);
+            }
+            break;
+        default:
+            result = 0;
+        }
+    }
+    Py_END_ALLOW_THREADS
+
+    path_cleanup(&_path);
+    if (result) {
+        Py_RETURN_TRUE;
+    }
+    Py_RETURN_FALSE;
+}
+
+
+/*[clinic input]
+os._exists
+
+    path: 'O'
+
+Return True if path refers to an existing path or an open file descriptor.
+
+Return False for broken symbolic links. On some platforms, this function may
+return False if permission is not granted to execute os.stat() on the requested
+file, even if the path exists.
+
+[clinic start generated code]*/
+
+static PyObject *
+os__exists_impl(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=0f0ba589f386f816 input=4fb71ae480a2082c]*/
+{
+    HANDLE hfile;
+    BOOL close_file = TRUE;
+    path_t _path = PATH_T_INITIALIZE("exists", "path", 0, 1);
+    int result;
+
+    if (!path_converter(path, &_path)) {
+        path_cleanup(&_path);
+        if (PyErr_ExceptionMatches(PyExc_ValueError)) {
+            PyErr_Clear();
+            Py_RETURN_FALSE;
+        }
+        return NULL;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    if (_path.fd != -1) {
+        hfile = _Py_get_osfhandle_noraise(_path.fd);
+        close_file = FALSE;
+    }
+    else {
+        hfile = CreateFileW(_path.wide, FILE_READ_ATTRIBUTES, 0, NULL,
+                            OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, NULL);
+    }
+    if (hfile != INVALID_HANDLE_VALUE) {
+        result = 1;
+        if (close_file) {
+            CloseHandle(hfile);
+        }
+    }
+    else {
+        STRUCT_STAT st;
+        switch (GetLastError()) {
+        case ERROR_ACCESS_DENIED:
+        case ERROR_SHARING_VIOLATION:
+        case ERROR_CANT_ACCESS_FILE:
+        case ERROR_INVALID_PARAMETER:
+            if (STAT(_path.wide, &st)) {
+                result = 0;
+            }
+            else {
+                result = 1;
+            }
+            break;
+        default:
+            result = 0;
+        }
+    }
+    Py_END_ALLOW_THREADS
+
+    path_cleanup(&_path);
+    if (result) {
+        Py_RETURN_TRUE;
+    }
+    Py_RETURN_FALSE;
+}
+
+
+/*[clinic input]
+os._islink
+
+    path: 'O'
+
+Return True if path refers to an existing directory entry that is a symbolic link.
+
+Return False if symbolic links are not supported by the Python runtime.
```

Comment:
```suggestion
Test whether a path is a symbolic link
```

The old docstring of `ntpath.islink()` is outdated. Windows XP (NT 5.1) hasn't been supported since Python 3.4, so there is no reason to mention the requirement of NT 6.0 or later. 

The definition of Python `ntpath.islink()` can be removed if the definition of `posixpath.islink()` is moved to `genericpath`. That should have been done a long time ago.